modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/sqlserver/SQLServerTypeMapper.java

```diff
@@ -0,0 +1,204 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.gce.imagemosaic.catalog.sqlserver;
+
+import java.util.*;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.data.transform.Definition;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.geotools.gce.imagemosaic.catalog.oracle.DataStoreWrapper;
+import org.geotools.gce.imagemosaic.catalog.oracle.FeatureTypeMapper;
+import org.geotools.jdbc.JDBCDataStore;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.AttributeType;
+import org.opengis.feature.type.GeometryType;
+import org.opengis.feature.type.Name;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+/**
+ * Specific SQLServer implementation for a {@link DataStoreWrapper} By default, SQLServer
+ * identifiers can't be longer than 128 chars. See <a
+ * href="https://docs.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver15">SQL
+ * Syntax identifiers</a>
+ */
+public class SQLServerTypeMapper implements FeatureTypeMapper {
```

Comment:
see my comment at the end of this PR about eventually moving duplicated code on a superclass.

Reply:
fixed